apps/emqx_ds_builtin_local/src/emqx_ds_builtin_local.erl

```diff
@@ -314,7 +326,53 @@ update_iterator(DB, Iter0 = #{?tag := ?IT, ?shard := Shard, ?enc := StorageIter0
 
 -spec next(emqx_ds:db(), iterator(), pos_integer()) -> emqx_ds:next_result(iterator()).
 next(DB, Iter, N) ->
-    with_worker(do_next, [DB, Iter, N]).
+    {ok, Ref} = emqx_ds_lib:with_worker(undefined, ?MODULE, do_next, [DB, Iter, N]),
+    receive
+        #poll_reply{ref = Ref, payload = Data} ->
+            Data
+    end.
+
+-spec poll(emqx_ds:db(), emqx_ds:poll_iterators(), emqx_ds:poll_opts()) -> {ok, reference()}.
+poll(DB, Iterators, PollOpts = #{timeout := Timeout}) ->
+    %% Create a new alias, if not already provided:
+    case PollOpts of
+        #{reply_to := ReplyTo} ->
+            ok;
+        _ ->
+            ReplyTo = alias([explicit_unalias])
+    end,
+    %% Spawn a helper process that will notify the caller when the
+    %% poll times out:
+    _Completion = spawn_link(
+        fun() ->
+            wait_completion(ReplyTo, Timeout)
+        end
+    ),
```

Comment:
This `wait_completion/2` apparently is never cancelled when (if) a reply containing data is received, so the same alias could receive both a reply with data and a `poll_timeout`.  

Was the original intent to use `alias([reply])`, so that contradictory responses to the same poll won't arrive:

```suggestion
            ReplyTo = alias([reply])
    end,
    %% Spawn a helper process that will notify the caller when the
    %% poll times out:
    _Completion = spawn_link(
        fun() ->
            wait_completion(ReplyTo, Timeout)
        end
    ),
```
?

Reply:
No, `reply` aliases can't be used here at all, they will cause all responses except the first one to be lost.
Currently name `wait_completion` is slightly misleading, as this helper process will fire timeout event unconditionally (there's no easy & cheap way to cancel it, given that requests will be routed to different nodes).

Reply:
I see.  I thought that one poll request meant at most one response back, but there might be several responses if there are multiple iterators.